sda-commons-starter/src/main/java/org/sdase/commons/starter/SdaPlatformBundle.java

```diff
@@ -0,0 +1,346 @@
+          corsBundleBuilder,
+          consumerTokenBundleBuilder,
+          openApiBundleBuilder);
+    }
+
+    // InitialBuilder
+
+    @Override
+    public ConsumerTokenConfigBuilder<SdaPlatformConfiguration> usingSdaPlatformConfiguration() {
+      return usingCustomConfig(SdaPlatformConfiguration.class)
+          .withAuthConfigProvider(SdaPlatformConfiguration::getAuth)
+          .withCorsConfigProvider(SdaPlatformConfiguration::getCors);
+    }
+
+    @Override
+    public <T extends Configuration> AuthConfigProviderBuilder<T> usingCustomConfig(
+        Class<T> configurationClass) {
+      return new InitialBuilder<>();
+    }
+
+    // CustomConfigurationProviders and follow up configuration providers
+
+    @Override
+    public CorsConfigProviderBuilder<C> withoutAuthentication() {
+      return this;
+    }
+
+    @Override
+    public CorsConfigProviderBuilder<C> withAuthConfigProvider(
+        AuthConfigProvider<C> authConfigProvider) {
+      this.authBundleBuilder =
+          AuthBundle.builder()
+              .withAuthConfigProvider(authConfigProvider)
+              .withAnnotatedAuthorization();
+      return this;
+    }
+
+    @Override
+    public CorsConfigProviderBuilder<C> withOpaAuthorization(
+        AuthConfigProvider<C> authConfigProvider, OpaConfigProvider<C> opaConfigProvider) {
+      this.authBundleBuilder =
+          AuthBundle.builder()
+              .withAuthConfigProvider(authConfigProvider)
+              .withExternalAuthorization();
+      this.opaBundleBuilder = OpaBundle.builder().withOpaConfigProvider(opaConfigProvider);
+      return this;
+    }
+
+    @Override
+    public ConsumerTokenConfigBuilder<C> withoutCorsSupport() {
+      return this;
+    }
+
+    @Override
+    public ConsumerTokenConfigBuilder<C> withCorsConfigProvider(
+        CorsConfigProvider<C> corsConfigProvider) {
+      this.corsBundleBuilder = CorsBundle.builder().withCorsConfigProvider(corsConfigProvider);
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withoutConsumerTokenSupport() {
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withOptionalConsumerToken() {
+      consumerTokenConfig = new ConsumerTokenConfig();
+      consumerTokenConfig.setOptional(true);
+      this.consumerTokenBundleBuilder =
+          ConsumerTokenBundle.builder().withConfigProvider(c -> consumerTokenConfig);
+      return this;
+    }
+
+    @Override
+    public ConsumerTokenRequiredConfigInitialBuilder<C> withRequiredConsumerToken() {
+      consumerTokenConfig = new ConsumerTokenConfig();
+      consumerTokenConfig.setOptional(false);
+      this.consumerTokenBundleBuilder =
+          ConsumerTokenBundle.builder().withConfigProvider(c -> consumerTokenConfig);
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withConsumerTokenConfigProvider(
+        ConsumerTokenConfigProvider<C> consumerTokenConfigProvider) {
```

Comment:
Open for discussion: When I first used the `ConsumerToken` configuration in the starter bundle I was wondering why there is no `consumerToken` in the configuration until i understood that this consumer token configuration is just for making consumer tokens optional or required for incoming requests. I think the naming of the configuration is quite confusing since the name is `withConsumerTokenConfigProvider(SdaPlatformConfiguration::getConsumerToken)`. When I read `getConsumerToken` then I would expect the actual consumer token and not a configuration. Furthermore if you would like to define a consumer token for outgoing calls, you have to name it like 'consumerTokenName'. I would vote for a renaming here to make clear that this configuration is for incoming requests.

Reply:
I've also heard from different people that the name is not intuitively clear.

Although the name is used according to other config properties. E.g. the mongo configuration is mostly added as "mongo".

What would be a better name? `consumerTokenSettings`? `consumerTokenServer`?

Reply:
The consumer token or consumer name seems like a workaround since I'm working for the SDA. At the moment it is only used for logging. I wonder if it is needed to set it technically as required in a service while it is a (temporary?) required pattern for the platform in general that may change when we introduce ISTIO.
It was meant to require identification of the calling service which should be part of the authorization we do in OPA. Actually it is nothing more than an indicator of who is calling as it can't be verified. I know moving that decision to OPA and not requiring it in the service may have a big impact on existing services but gives more flexibility for the future. I'd like to hear about @Fox32 opinion. I think he has quite a lot of insights about the requirements of the consumer token, the use in current services at SDA and at developing customers and about the future evolution of service authentication in context of a service mesh. Maybe @dhenneke can also point out some arguments about moving such a decision to OPA.

Reply:
My main concern with `withConsumerTokenConfigProvider(...)` is that we have two options to configure the consumer token: You can use `withoutConsumerTokenSupport()` / `withRequiredConsumerToken()` / `withOptionalConsumerToken()` OR the `withConsumerTokenConfigProvider(...)`. My current understanding of the consumer token (and I guess the majority selection) is that the service should decide whether it want to require a consumer token during build-time. We shouldn't give the users of the `SdaPlatformBundle` the choice between deciding it in code vs. deciding it in config, regardless of the decision whether it should be optional or required. Especially since there is no guidance on what the preferred way is. (And I agree that it is very easy to confuse the server-setting and the client-setting in a configuration).

About the need, our wiki cites:
> A consumer token is used to identify the consumer of a service. This is used to aggregate metrics by consumer, to filter logs, and for billing purposes. The consumer token should be provided with every request in form of the `Consumer-Token` request header. If no consumer token is provided, the access is denied with status `401 - UNAUTHORIZED`.

It's not really an actual authentication measure but more a technical/organizational guideline. In this regard, maybe the `401` was not the best choice, but that shouldn't be discussed in this PR.

Reply:
So your conclusion / recommendation is to remove `withConsumerTokenConfigProvider`?

Reply:
Or remove the other three? Or we decide that this needs a deeper look and decide it in the future.

Reply:
As @dhenneke already said: Consumer-Token is only used for logging and metrics. So more or less a simple version of tracing. Maybe we could even make it completely optional and not require it at all. I even wonder how this works well with customers that don't use the consumer token in the infrastructure. That is probably also the reason for making it configurable at deployment time. Otherwise I would vote for making it an implementation time decision - so remove the option. However I'm not sure if the customers will like that.

Reply:
> Maybe we could even make it completely optional and not require it at all.

That would mean, we make _withOptionalConsumerToken()_ the default and remove all other options?

I would like that approach. We could add the other options if some requests for it.

Reply:
Please take a look at [PR-662](https://github.com/SDA-SE/sda-dropwizard-commons/pull/662)